lib/src/sidenav/Sidenav.tsx

```diff
@@ -1,56 +1,148 @@
+    <path d="M0 0h24v24H0z" fill="none" />
+    <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
+  </svg>
+);
+
+const collapsedIcon = (
+  <svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="currentColor">
+    <path d="M0 0h24v24H0z" fill="none" />
+    <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
+  </svg>
+);
+
+const externalLinkIcon = (
+  <svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24" fill="currentColor">
+    <path d="M0 0h24v24H0z" fill="none" />
+    <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z" />
+  </svg>
+);
 
+const DxcSidenav = ({ children }: SidenavPropsType): JSX.Element => {
+  const colorsTheme = useTheme();
   return (
     <ThemeProvider theme={colorsTheme.sidenav}>
-      <SidenavContainer padding={padding}>
-        <BackgroundColorProvider color={colorsTheme.sidenav.backgroundColor}>{children}</BackgroundColorProvider>
+      <SidenavContainer>
+        <BackgroundColorProvider color={colorsTheme.sidenav.backgroundColor}>
+          {React.Children.map(children, (child: ReactElement<any>) => {
+            if (child.type === Title) {
+              return child;
+            }
+          })}
+          <DxcStack divider={true} gutter="small">
+            {React.Children.map(children, (child: ReactElement<any>) => {
+              if (child.type === Section) {
+                return child;
+              }
+            })}
+          </DxcStack>
+        </BackgroundColorProvider>
       </SidenavContainer>
     </ThemeProvider>
   );
 };
 
-const Title = ({ children }: SidenavTitlePropsType): JSX.Element => <SidenavMenuTitle>{children}</SidenavMenuTitle>;
-
-const Subtitle = ({ children }: SidenavSubtitlePropsType): JSX.Element => (
-  <SidenavMenuSubTitle>{children}</SidenavMenuSubTitle>
+const Title = ({ children, icon }: SidenavTitlePropsType): JSX.Element => (
+  <SidenavTitle>
+    {icon}
+    {children}
+  </SidenavTitle>
 );
 
-const Link = ({ tabIndex = 0, href, onClick, children }: SidenavLinkPropsType): JSX.Element => {
-  const setIsSidenavVisibleResponsive = useResponsiveSidenavVisibility();
-  const handleClick = () => {
-    onClick?.();
-    setIsSidenavVisibleResponsive?.(false);
-  };
+const Section = ({ children }: SidenavSectionPropsType): JSX.Element => <DxcStack gutter="small">{children}</DxcStack>;
 
+const Group = ({ children, title, collapsable = false, icon }: SidenavGroupPropsType): JSX.Element => {
+  const [collapsed, setCollapsed] = useState(false);
   return (
-    <SidenavMenuLink tabIndex={tabIndex} href={href} onClick={handleClick}>
-      {children}
-    </SidenavMenuLink>
+    <SidenavGroup>
+      {collapsable && title ? (
+        <SidenavGroupTitleButton aria-expanded={collapsed} onClick={() => setCollapsed(!collapsed)}>
+          <span>
```

Comment:
This must be a styled component. `SidenavGroupTitle` and this span share the same CSS rules, so it can be reused here and removed the common CSS rules from `SidenavGroup` and its `span {}` selector, padding, box-sizing, font rules, etc.

Reply:
I already tried that scenario and unfortunately, because of the presence of the collapsable indicator, it does not work correctly sharing the same css. The few similarities are only in the font but they also affect the collapsable indicator.

Reply:
Thinking more regarding this I think I understand a bit more about this and I have also applied it to the linkContent mentioned below.